### PR TITLE
Validators in SMT

### PR DESCRIPTION
**Pull request by NirnayaSindhuSuthari**
Fixes [b/295990119](http://b/295990119)

-Added Validators on SMT Database name and Index name.
-Added regex, icon, message and link of naming convention in Database form.
-Added regex, message and link of naming convention in Index form.
-Modified count Checker

-Adding screenshots
Screenshot from 2024-06-10 05-37-10
Screenshot from 2024-06-10 05-38-22

### CODE REVIEW

File: ui/src/app/components/add-index-form/add-index-form.component.scss [hunk 1]
```diff
@@ -2,7 +2,6 @@
   width: 100%;
   padding: 0px;
 }
-
```
Comment (asthamohta) [inline ui/src/app/components/add-index-form/add-index-form.component.scss:5]: nit: remove this file

File: ui/src/app/components/target-details-form/target-details-form.component.scss [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.mat-mdc-form-field{
```
Comment (asthamohta) [inline ui/src/app/components/target-details-form/target-details-form.component.scss:1]: How is this different from the default css?

Comment (NirnayaSindhuSuthari) [reply ui/src/app/components/target-details-form/target-details-form.component.scss:1]: -If this is not written.

Screenshot from 2024-06-10 10-20-10

-It makes the textbox width to full if I include this.

File: ui/src/app/components/add-index-form/add-index-form.component.scss (hunk 1 above)
Comment (NirnayaSindhuSuthari) [reply ui/src/app/components/add-index-form/add-index-form.component.scss:5]: -ok.

**Review by NirnayaSindhuSuthari — COMMENTED** [review]
Done with changes required.

File: ui/src/app/components/target-details-form/target-details-form.component.scss (hunk 2 above)
Comment (asthamohta) [reply ui/src/app/components/target-details-form/target-details-form.component.scss:1]: Understood, looks good

**Review by asthamohta — APPROVED** [review]
LGTM! Congratulations on your first merge

**Pull request merged**